### PR TITLE
Add ManagedBuildManager#createConfigurationForProject()

### PR DESCRIPTION
**Pull request by 15knots**
This should allow ISV's to create MBS based project with a vendor-specific build-system ID without using internal API.

Signed-off-by: 15knots <11367029+15knots@users.noreply.github.com>

### CODE REVIEW

**Comment by 15knots** [comment]
Sorry, forgat the N&N.... Coming soon.

File: build/org.eclipse.cdt.managedbuilder.ui/src/org/eclipse/cdt/managedbuilder/ui/wizards/MBSWizardHandler.java [hunk 1]
```diff
@@ -611,20 +610,15 @@ private void setProjectDescription(IProject project, boolean defaults, boolean o
...
-			String id = ManagedBuildManager.calculateChildId(cf.getId(), null);
-			Configuration config = new Configuration(mProj, cf, id, false, true);
-			CConfigurationData data = config.getConfigurationData();
-			ICConfigurationDescription cfgDes = des.createConfiguration(ManagedBuildManager.CFG_DATA_PROVIDER_ID, data);
-			config.setConfigurationDescription(cfgDes);
-			config.exportArtifactInfo();
-
+			IConfiguration config = ManagedBuildManager.createConfigurationForProject(des, mProj, cf,
+					ManagedBuildManager.CFG_DATA_PROVIDER_ID);
+			// TODO check 			config.exportArtifactInfo();
```
Comment (mbooth101) [inline build/org.eclipse.cdt.managedbuilder.ui/src/org/eclipse/cdt/managedbuilder/ui/wizards/MBSWizardHandler.java:621]: Looks like this call to exportArtifactInfo() was moved to createConfigurationForProject()

What is TODO here?

**Review by jonahgraham — COMMENTED** [review]
LGTM - Thanks for recreating a clean commit. Thanks @mbooth101 for doing a review too.

File: build/org.eclipse.cdt.managedbuilder.ui/src/org/eclipse/cdt/managedbuilder/ui/wizards/MBSWizardHandler.java (hunk 1 above)
Comment (15knots) [reply build/org.eclipse.cdt.managedbuilder.ui/src/org/eclipse/cdt/managedbuilder/ui/wizards/MBSWizardHandler.java]: Forgot to delete that. It is a reminder to check whether the refactored code should call 'config.exportArtifactInfo();'.

**Pull request merged**